core/templates/dev/head/pages/exploration-editor-page/services/learner-answer-details-data.service.ts

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to get learner answer info data and to delete
+ * any learner answer info.
+ */
+
+require('pages/exploration-editor-page/services/exploration-data.service.ts');
+require('domain/utilities/UrlInterpolationService.ts');
+require('domain/statistics/LearnerAnswerInfoObjectFactory.ts');
+require('domain/statistics/LearnerAnswerDetailsObjectFactory.ts');
+
+angular.module('oppia').factory('LearnerAnswerDetailsDataService', [
+  '$http', '$q', 'ExplorationDataService', 'LearnerAnswerDetailsObjectFactory',
+  'LearnerAnswerInfoObjectFactory', 'UrlInterpolationService',
+  function(
+      $http, $q, ExplorationDataService, LearnerAnswerDetailsObjectFactory,
+      LearnerAnswerInfoObjectFactory, UrlInterpolationService) {
+    var _expId = ExplorationDataService.explorationId;
+    var _data = [];
+    var learnerAnswerInfoData = null;
+    var LEARNER_ANSWER_INFO_DATA_URL = (
+      '/learneranswerinfohandler/learner_answer_details/<entity_type>' +
+      '/<entity_id>');
+
+    var _fetchLearnerAnswerInfoData = function(
+        successCallback, errorCallback) {
+      var learnerAnswerInfoDataUrl = UrlInterpolationService.interpolateUrl(
+        LEARNER_ANSWER_INFO_DATA_URL, {
+          entity_type: 'exploration',
+          entity_id: _expId});
+
+      $http.get(learnerAnswerInfoDataUrl).then(function(response) {
```

Comment:
Return this value, it will be a `Promise` instance.

Reply:
Done